types/dygraphs/index.d.ts

```diff
@@ -1016,24 +1253,27 @@ declare class Dygraph {
      *
      * Note: use toDomXCoord instead of toDomCoords(x, null) and use toDomYCoord
      * instead of toDomCoords(null, y, axis).
+     * {@link https://dygraphs.com/jsdoc/symbols/Dygraph.html#toDomCoords}
      */
-    toDomCoords(x: number, y: number, axis?: number): [number, number];
+    toDomCoords<X extends number | null, Y extends number | null>(x: X, y: Y, axis?: number): [X, Y];
 
     /**
      * Convert from data x coordinates to canvas/div X coordinate.
      * If specified, do this conversion for the coordinate system of a particular
      * axis.
      * Returns a single value or null if x is null.
+     * {@link https://dygraphs.com/jsdoc/symbols/Dygraph.html#toDomXCoord}
      */
-    toDomXCoord(x: number): number;
+    toDomXCoord<T extends number | null>(x: T): T;
 
     /**
-     * Convert from data y coordinates to canvas/div Y coordinate and optional
+     * Convert from data x coordinates to canvas/div Y coordinate and optional
      * axis. Uses the first axis by default.
      *
-     * returns a single value or null if y is null.
+     * Returns a single value or null if y is null.
+     * {@link https://dygraphs.com/jsdoc/symbols/Dygraph.html#toDomYCoord}
      */
-    toDomYCoord(y: number, axis?: number): number;
+    toDomYCoord<T extends number | null>(y: T, axis?: number): T;
```

Comment:
Is the goal here to have the return type include `null` only if the input type is `null`? If so, the right way to do that is with a conditional type:

```ts
toDomYCoord<T extends number | null>(y: T, axis?: number): T extends number ? number : null;
```

Reply:
Thanks, updated